cdap/resource_streaming_program.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdap
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// https://docs.cdap.io/cdap/current/en/reference-manual/http-restful-api/lifecycle.html.
+func resourceStreamingProgram() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceStreamingProgramCreate,
+		Read:   resourceStreamingProgramRead,
+		Delete: resourceStreamingProgramDelete,
+		Exists: resourceStreamingProgramExists,
+
+		Schema: map[string]*schema.Schema{
+			"namespace": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of the namespace in which this resource belongs. If not provided, the default namespace is used.",
+				DefaultFunc: func() (interface{}, error) {
+					return defaultNamespace, nil
+				},
+			},
+			"app": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the application being called.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the flow, MapReduce, custom service, Spark, worker, or workflow being called.",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "One of flows, mapreduce, services, spark, workers, or workflows.",
+			},
+			"runtime_arguments": {
```

Comment:
Why not just a map?

Reply:
So one of the primary usecases for these runtime args is by setting 
a key `system.profile.name`. I thought that having a `.` in the key might might be an issue.
I agree that map makes more sense. 
I will test if dots are allowed.

Reply:
right now I notice I'm not actually reconstructing the correct payload (everything wilbe `[{key: ..., val: ...}]` which is not correct.